string posename
---
int64 res
